internal/org/repository/eventsourcing/eventstore.go

```diff
@@ -0,0 +1,187 @@
+package eventsourcing
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/caos/zitadel/internal/errors"
+	"github.com/caos/zitadel/internal/eventstore"
+	es_models "github.com/caos/zitadel/internal/eventstore/models"
+	es_sdk "github.com/caos/zitadel/internal/eventstore/sdk"
+	org_model "github.com/caos/zitadel/internal/org/model"
+)
+
+type OrgEventstore struct {
+	eventstore.Eventstore
+}
+
+type OrgConfig struct {
+	eventstore.Eventstore
+}
+
+func StartOrg(conf OrgConfig) *OrgEventstore {
+	return &OrgEventstore{Eventstore: conf.Eventstore}
+}
+
+func (es *OrgEventstore) PrepareCreateOrg(ctx context.Context, orgModel *org_model.Org) (*Org, []*es_models.Aggregate, error) {
+	if orgModel == nil || !orgModel.IsValid() {
+		return nil, nil, errors.ThrowInvalidArgument(nil, "EVENT-OeLSk", "org not valid")
+	}
+	id, err := idGenerator.NextID()
+	if err != nil {
+		return nil, nil, errors.ThrowInternal(err, "EVENT-OwciI", "id gen failed")
+	}
+	orgModel.AggregateID = strconv.FormatUint(id, 10)
+	org := OrgFromModel(orgModel)
+
+	aggregates, err := orgCreatedAggregates(ctx, es.AggregateCreator(), org)
+
+	return org, aggregates, err
+}
+
+func (es *OrgEventstore) CreateOrg(ctx context.Context, orgModel *org_model.Org) (*org_model.Org, error) {
+	org, aggregates, err := es.PrepareCreateOrg(ctx, orgModel)
+	err = es_sdk.PushAggregates(ctx, es.PushAggregates, org.AppendEvents, aggregates...)
+	if err != nil {
+		return nil, err
+	}
+
+	return OrgToModel(org), nil
+}
+
+func (es *OrgEventstore) OrgByID(ctx context.Context, org *org_model.Org) (*org_model.Org, error) {
+	if org == nil {
+		return nil, errors.ThrowInvalidArgument(nil, "EVENT-gQTYP", "org not set")
+	}
+	query, err := OrgByIDQuery(org.AggregateID, org.Sequence)
+	if err != nil {
+		return nil, err
+	}
+
+	esOrg := OrgFromModel(org)
+	err = es_sdk.Filter(ctx, es.FilterEvents, esOrg.AppendEvents, query)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+	if esOrg.Sequence == 0 {
+		return nil, errors.ThrowNotFound(nil, "EVENT-kVLb2", "org not found")
+	}
+
+	return OrgToModel(esOrg), nil
+}
+
+func (es *OrgEventstore) DeactivateOrg(ctx context.Context, orgModel *org_model.Org) (*org_model.Org, error) {
```

Comment:
in my opinion we should just pass orgid here and read existing, because we want to deactivate actual org

Reply:
done